src/engines/nwn2/inventory.cpp

```diff
@@ -83,6 +83,26 @@ Item *Inventory::getNextItemInInventory() {
 	return 0;
 }
 
+Item *Inventory::createItem(const Common::UString &blueprint, uint16 stackSize, const Common::UString &tag) {
+	if (blueprint.empty() || stackSize == 0)
+		return 0;
+
+	// Create the item
+	Item *item = new Item(blueprint, stackSize, tag);
+	if (item) {
```

Comment:
Can you rewrite this method to be less nested, by returning directly? I.e.

```
if (!item)
	return 0;
```

etc.

Reply:
It `Item` has no overrided `operator new`, then checking for `nullptr` is not required -- an exception will be thrown if memory can not be allocated

Reply:
Oh, I didn't see that it's just a plain new and not a function() call. My bad.

(However, whether an exception is thrown is optional. The operating system might simply overcommit. But yes, in either case, checking for nullptr is useless.)